src/main/java/org/truffleruby/core/kernel/TruffleKernelNodes.java

```diff
@@ -81,7 +81,7 @@ public boolean load(VirtualFrame frame, DynamicObject file, boolean wrap,
 
             try {
                 final RubyRootNode rootNode = getContext().getCodeLoader().parse(
-                        getContext().getSourceLoader().load(StringOperations.getString(file)),
+                        getContext().getSourceLoader().load(StringOperations.getString(file).intern()),
```

Comment:
I think the SourceLoader should intern when passing it as the name of the Source instead.
The String passed to SourceLoader.load() is not necessarily passed to the Source builder and shouldn't gain from being interned.

Reply:
Thanks; it does already, SourceLoader:194 (`name.intern()`).

Reply:
(in fact, that's another change of me in this PR; so I will just remove the intern() when calling `SourceLoader.load()`)